File: 프로그래머스/3/12971. 스티커 모으기（2）/스티커 모으기（2）.js
function solution(sticker) {
    var answer = 0;
    
    if (sticker.length === 1){
        return sticker[0]
    }
    
    if (sticker.length === 2){
        return Math.max(sticker[0], sticker[1])
    }
    
    const withOne = sticker.slice(0, sticker.length-1)
    const withOutOne = sticker.slice(1, sticker.length)
    
    const dp = [withOne[0], Math.max(withOne[0], withOne[1])]
    
    const dp2 = [withOutOne[0], Math.max(withOutOne[0], withOutOne[1])]
    
    for(let i=2; i<withOne.length; i++){
        dp[i] = Math.max(dp[i-2] + withOne[i], dp[i-1])
        dp2[i] = Math.max(dp2[i-2] + withOutOne[i], dp2[i-1])
    }
    
    

    return Math.max(dp.at(-1), dp2.at(-1));
}

File: 프로그래머스/3/12971. 스티커 모으기（2）/스티커 모으기（2）.py
def solution(sticker):
    answer = 0
    
    if len(sticker) == 1:
        return sticker[0]
    elif len(sticker) == 2:
        return max(sticker[0], sticker[1])
    
    withOne = []
    withOutOne = []
    
    for i in range(0, len(sticker)-1):
        withOne.append(sticker[i])
        
    for i in range(1, len(sticker)):
        withOutOne.append(sticker[i])
    
    
    
    dp = [withOne[0], max(withOne[0], withOne[1])]
    dp2 = [withOutOne[0], max(withOutOne[0], withOutOne[1])]
    
    
    for i in range(2, len(sticker)-1):
        dp.append(max(dp[i-2] + withOne[i], dp[i-1]))
        dp2.append(max(dp2[i-2] + withOutOne[i], dp2[i-1]))

    
    return max(dp[-1], dp2[-1])